bundles/framework/rpc/service/RpcService.js

```diff
@@ -0,0 +1,140 @@
+
+/**
+ * @class Oskari.mapframework.bundle.rpc.service.RpcService
+ * Allows extend RPC supported funtions
+ *
+ */
+Oskari.clazz.define('Oskari.mapframework.bundle.rpc.service.RpcService',
+
+    /**
+     * @method create called automatically on construction
+     * @static
+     *
+     */
+
+    function (instance) {
+        this.instance = instance;
+        this.conf = this.instance.conf || {};
+        this.channel = this.instance._channel;
+        this._log = Oskari.log('RpcService');
+        this._availableFunctions = {};
+        this._allowedFunctions = this.conf.allowedfunctions || [];
+        this._allowedFunctionsFromConf = Array.isArray(this.conf.allowedfunctions);
+    }, {
+        __name: 'Rpc.RpcService',
+        __qname: 'Oskari.mapframework.bundle.rpc.service.RpcService',
+
+        getQName: function () {
+            return this.__qname;
+        },
+
+        getName: function () {
+            return this.__name;
+        },
+
+        /**
+         * @method init
+         * Initializes the service
+         */
+        init: function () {
+        },
+
+        /**
+         * @method addFunction
+         * Add function to RPC
+         *
+         * @param {String} name function name
+         * @param {Function} func function that should be added
+         */
+        addFunction: function (name, func) {
+            if (!func && typeof func !== 'function') {
+                return;
+            }
+
+            if (this._availableFunctions[name] != null) {
+                this._log.warn('Trying add already defined RPC function (' + func.name + '), skipping.');
+                return;
+            }
+
+            this._availableFunctions[name] = func;
+
+            // bind functions
+            this.__updateAllowedFunctions(name);
+            this.__bindFunction(name);
+        },
+
+        /**
+         * @public @method getAllowedFunctions
+         * Get allowed functions
+         *
+         * @returns {Object} allowed functions
+         */
+        getAllowedFunctions: function () {
+            return arrayToObject(this._allowedFunctions);
+        },
+
+        /**
+         * @private @method __bindFunction
+         * Bind allowed function to channel
+         *
+         * @param {String} name function name
+         */
+        __bindFunction: function (name) {
+            const me = this;
+
+            if (!me._availableFunctions[name]) {
+                return;
+            }
+
+            me.channel.bind(name, function (trans, params) {
+                if (!domainMatch(trans.origin, me.conf.domain)) {
+                    // eslint-disable-next-line no-throw-literal
+                    throw {
+                        error: 'invalid_origin',
+                        message: 'Invalid origin: ' + trans.origin
+                    };
+                }
+
+                const requestedFunction = me._availableFunctions[name];
```

Comment:
We can store reference to the function at the beginning so we don't have to reference _availableFunctions twice.
```suggestion

            const requestedFunction = me._availableFunctions[name];
            if (!requestedFunction) {
                return;
            }

            me.channel.bind(name, function (trans, params) {
                if (!domainMatch(trans.origin, me.conf.domain)) {
                    // eslint-disable-next-line no-throw-literal
                    throw {
                        error: 'invalid_origin',
                        message: 'Invalid origin: ' + trans.origin
                    };
                }

```